contracts/registry/implementation/IdentityRegistry.sol

```diff
@@ -205,7 +205,7 @@ contract IdentityRegistry is IIdentityRegistry, AgentRoleUpgradeable, IRStorage
             if (claimIds.length == 0) {
                 return false;
             }
-            require(claimIds.length <= 10, 'too much claims of same topic');
+            require(claimIds.length < 10, 'too much claims of same topic');
```

Comment:
This one is important to note:
- A user with too much claims of the same topic will not be able to receive TREX tokens.
- The transaction would be reverted, and the user would have to remove some of their claims.

It's possible that many claims of the same topics exists on an identity. Let's say we have a claim of topic "YOU ARE OVER 18", then many issuers could emit such a claim.

Maybe we should think of another way to query claims, as follow:
- At this point in the contract, we know the topic (for `claimTopic` in `requiredClaimTopics`).
- We can query the list of claim topics associated with a trusted issuer (but I don't think we can query the list of trusted issuers for a given claim topic, we could optimize there).
- For each claim topic, we compute the claim ID with the trusted claim issuers.
- We can query on the identity contract each one of these computed claim ID.

Instead of querying an array of claims (that may be issued by untrusted claim issuers), we would only query claim of topic emitted by trusted issuers for this specific topic.

I can implement something like this to see how it goes.

Reply:
indeed, knowing that the claim ID is `bytes32 claimId = keccak256(abi.encode(_issuer, _topic))` we could check it this way instead of searching in an array of claims, it would cost less gas probably

Reply:
After checking closer, it is not better, or we should modify the TIR to have a function to get the list of trusted issuers for a given claim topic, otherwise you will need to call `hasClaimTopic` on each trusted issuer for each claim topic required, to calculate the claimID and look if the OID owner has that claimID and if it is valid and not revoked. 
Probably more costly than the actual implementation. 
If we want to go this way we need to add a variable `mapping(uint256 => address[]) trustedIssuers` and a function `getTrustedIssuers(uint256 _claimTopic)` ... It is not complicated to do, we can do it maybe... not sure if it would really save gas or not

Reply:
In #129 I will implement the first optimization (computing the claim ID to query only claims from trusted issuers).
I might also implement a `getTrustedIssuersByClaimTopic(uint256 claimTopic)` method in the Trusted Issuers Registry.